tools/pythonpkg/src/vector_conversion.cpp

```diff
@@ -151,13 +151,34 @@ void VectorConversion::NumpyToDuckDB(PandasColumnBindData &bind_data, py::array
 		}
 		break;
 	}
-	case PandasType::VARCHAR: {
+	case PandasType::VARCHAR:
+	case PandasType::OBJECT: {
 		auto src_ptr = (PyObject **)numpy_col.data();
 		auto tgt_ptr = FlatVector::GetData<string_t>(out);
 		auto &out_mask = FlatVector::Validity(out);
 		for (idx_t row = 0; row < count; row++) {
 			auto source_idx = offset + row;
-			auto val = src_ptr[source_idx];
+			py::str str_val;
+			PyObject *val = src_ptr[source_idx];
+#if PY_MAJOR_VERSION >= 3
+			if (bind_data.pandas_type == PandasType::OBJECT && !PyUnicode_CheckExact(val)) {
+				py::handle object_handle = val;
+				if (val == Py_None) {
+					out_mask.SetInvalid(row);
+					continue;
+				}
+				if (py::isinstance<py::float_>(val)) {
+					bool is_nan = py::cast<bool>(object_handle.attr("__ne__")(object_handle));
+					if (is_nan) {
+						out_mask.SetInvalid(row);
```

Comment:
NaN values don't appear to be tested. Is this special case here necessary in the first place since we are converting to strings anyway?

Reply:
Is this not being covered in the coverall? Where can I check that?

The reason for that code is that None for floats is represented as a NaN.
If I turn that into a string instead of keeping a null I get the string NaN, which then I can't verify nullity anymore.

Reply:
The Python extension is not included in the code coverage right now. I haven't tried adding it - I can have a go at that. 

> The reason for that code is that None for floats is represented as a NaN.

That is fair but I don't see this included in the test added with this PR. Is this tested elsewhere?

Reply:
Yes I actually added it because it was breaking some of the older tests.

If you add the codecov for python I'm happy to add tests to whatever is not covered.

Reply:
Fair enough.